dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/audit/AuditMessage.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.api.audit;
+
+import org.apache.dolphinscheduler.common.enums.AuditModuleType;
+import org.apache.dolphinscheduler.common.enums.AuditOperationType;
+import org.apache.dolphinscheduler.dao.entity.User;
+
+import java.util.Date;
+
+public class AuditMessage {
+    private User user;
+
+    private Date auditDate;
+
+    private AuditModuleType module;
+
+    private AuditOperationType operation;
+
+    private String projectName;
+
+    private String processName;
```

Comment:
The design of audit message looks bad, it includes the resources in it, so if we need to audit 100 kinds of resources (tenants, users, projects, workflows, workflow instances, ...), we have to put their `***Name` here? That would make the class explode and we have to modify the database schema over and over again, also there might be many fields empty (redundant) for many records.

In my mind, an audit log should only include an operation (say, CRUD), a resource type (for example, project, workflow), and a resource id (project id, workflow id), with that we can expand the audit logs easily

Reply:
Thanks for your review. 
Is it necessary to record the  operator information and the operation time ?  The resource id may include project id, workflow id ,datasource id and so on. The resouce id is a foreign key associated with different database tables.   It may confused for user to see the resource id on the UI interface.

Reply:
> Thanks for your review. 
> 
> Is it necessary to record the  operator information and the operation time ?  

Yes we need 

> The resource id may include project id, workflow id ,datasource id and so on. The resouce id is a foreign key associated with different database tables.   It may confused for user to see the resource id on the UI interface.

For resource id, I mean it's the id of the corresponding resource type. 

For example, when resource type is project, the resource id represents project id, when the resource is tenant, the resource id represents tenant id, etc. does it make sense to you ?

Reply:
OK, it makes sense. Thanks for you explanation.

My last question, is it ok to display information below in the UI? 

* operator
* operation time
* operation (CRUD)
* resource type 
* resource name (project name or workflow name ...)

Reply:
> 

I think the information listed above are good to me